util/src/main/java/org/datacommons/util/SummaryReportGenerator.java

```diff
@@ -70,10 +85,51 @@ public int getNumObservations() {
     public List<Double> getSeriesValues() {
       return this.seriesValues;
     }
-  }
 
-  public static boolean TEST_mode = false;
-  public static final String SUMMARY_REPORT_HTML = "summary_report.html";
+    public String getTimeSeriesChartSVG() {
+      TimeSeriesCollection dataset = new TimeSeriesCollection();
+      TimeSeries timeSeries = new TimeSeries("ts");
+      for (int i = 0; i < this.seriesDates.size(); i++) {
+        if (i >= this.seriesValues.size()) break;
+        LocalDateTime localDateTime = StringUtil.getValidISO8601Date(seriesDates.get(i));
+        if (localDateTime == null) continue;
+        Date date = Date.from(localDateTime.atZone(ZoneId.systemDefault()).toInstant());
+        timeSeries.addOrUpdate(new Day(date), this.seriesValues.get(i));
+      }
+      if (timeSeries.isEmpty()) return "";
+      dataset.addSeries(timeSeries);
+      JFreeChart chart = ChartFactory.createTimeSeriesChart("", "", "", dataset);
+      XYItemRenderer renderer = new XYLineAndShapeRenderer(true, true);
+      renderer.setSeriesShape(0, ShapeUtils.createDiamond(5));
+      renderer.setSeriesVisibleInLegend(0, false);
+      XYPlot plot = chart.getXYPlot();
+      plot.setBackgroundPaint(Color.WHITE);
+      plot.setRenderer(renderer);
+      if (timeSeries.findValueRange().getLength() == 0) {
+        // Manually set the range when the values in the time series are all the same. Otherwise,
+        // the chart library will draw an axis with multiple ticks all labeled with that same value.
+        plot.getRangeAxis().setRange(0, this.seriesValues.get(0) * 2);
+      }
+      if (this.seriesDates.get(0).equals(this.seriesDates.get(this.seriesDates.size() - 1))) {
```

Comment:
per comment, to check single data point case, why not do `this.seriesDates.size() == 1` ?

Reply:
Because this.seriesDate comes from the DataPoint type which can have multiple values for a single date, I assumed there could be duplicate dates in this.seriesDates. I double checked that there's actually no duplicate dates so you're right, I can just do this.seriesDates.size() == 1. Updated